lib/Component.php

```diff
@@ -346,6 +348,55 @@ function jsonSerialize() {
     }
 
     /**
+     * This method serializes the data into XML. This is used to create xCard or
+     * xCal documents.
+     *
+     * @param Xml\Writer $writer  XML writer.
+     * @return void
+     */
+    function xmlSerialize(Xml\Writer $writer) {
+
+        $components = [];
+        $properties = [];
+
+        foreach($this->children as $child) {
+            if ($child instanceof Component) {
+                $components[] = $child;
+            } else {
+                $properties[] = $child;
+            }
+        }
+
+        $writer->startElement(strtolower($this->name));
+
+        if (!empty($properties)) {
+
+            $writer->startElement('properties');
```

Comment:
This already exists: https://github.com/fruux/sabre-xml/blob/master/lib/Writer.php#L192-L200. Would you prefer this form?